### PR TITLE
Setup for Onboarding data source 

### PR DESCRIPTION
**Pull request by Sachinbisht27**
**ISSUE ID:** 
- https://github.com/DostEducation/whatsapp-webhook-analytics/issues/31

**Type of changes:**
- Corrected formatting of a few files.
- Added the changes to make the onboarding data source

### CODE REVIEW

File: models/staging/whatsapp-webhook/stg_users.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+with
+    source as (select * from {{ source("whatsapp_webhook_prod", "users") }}),
+
+    users as (
+
+        select
+            id as user_id,
+            name as user_name,
+            phone as user_phone,
+            location as user_location
```
Comment (Satendra-SR) [inline models/staging/whatsapp-webhook/stg_users.sql]: don't we need user's created on date for any type of analysis?

Comment (Satendra-SR) [reply models/staging/whatsapp-webhook/stg_users.sql]: Also, can't we just do it like - select * from users instead? this way, we don't have to add any new column individually in this transformation script

File: models/staging/whatsapp-webhook/scr_webhook.yml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 version: 2
 
 sources:
-  - name: webhook
+  - name: whatsapp_webhook_prod
     description: dataset created by whatsapp webhooks on BigQuery
     database: cryptic-gate-211900
-    schema: 'whatsapp_webhook_prod'
+    schema: "whatsapp_webhook_prod_v2"
```
Comment (Satendra-SR) [inline models/staging/whatsapp-webhook/scr_webhook.yml]: Let's revert this. The `whatsapp_webhook_prod` dataset is in US region now

File: models/intermediate/int_user_onboarding.sql [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{% set field_names_query %}
+    SELECT DISTINCT field_name FROM {{ ref('stg_user_attributes') }}
+{% endset %}
+
+with
+    user_attributes_cte as (
```
Comment (Satendra-SR) [inline models/intermediate/int_user_onboarding.sql]: what does `cte` means here?

File: models/intermediate/int_user_onboarding.sql [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                    end
+                ) as {{ field_name.field_name }}
+                {% if not loop.last %},{% endif %}
+            {% endfor %}
+        from {{ ref("stg_user_attributes") }}
+        group by user_id
+    )
+
+select gu.*, ua.* except(user_id)
+from {{ ref("stg_users") }} gu
```
Comment (Satendra-SR) [inline models/intermediate/int_user_onboarding.sql]: Can we have a better naming convention?

**Review by Satendra-SR — CHANGES_REQUESTED** [review]
@Sachinbisht27 added a few comments, have a look

File: models/intermediate/int_user_onboarding.sql (hunk 3 above)
Comment (Sachinbisht27) [reply models/intermediate/int_user_onboarding.sql]: CTE means Common Table Expression but here it doesn't make any sense so we can remove.

File: models/intermediate/int_user_onboarding.sql (hunk 4 above)
Comment (Sachinbisht27) [reply models/intermediate/int_user_onboarding.sql]: Sure

File: models/staging/whatsapp-webhook/scr_webhook.yml (hunk 2 above)
Comment (Sachinbisht27) [reply models/staging/whatsapp-webhook/scr_webhook.yml]: Sure

File: models/staging/whatsapp-webhook/stg_users.sql (hunk 1 above)
Comment (Sachinbisht27) [reply models/staging/whatsapp-webhook/stg_users.sql]: Sure we can take all the data from user table.

File: models/staging/whatsapp-webhook/scr_webhook.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 version: 2
 
 sources:
-  - name: webhook
```
Comment (Satendra-SR) [inline models/staging/whatsapp-webhook/scr_webhook.yml:4]: Can we keep it webhook only? or whatsapp_webhook? making it env specific may confuse in future

**Review by Satendra-SR — APPROVED** [review]
One last comment, else looks good to go.
Please fix and merge

File: models/staging/whatsapp-webhook/scr_webhook.yml (hunk 5 above)
Comment (Sachinbisht27) [reply models/staging/whatsapp-webhook/scr_webhook.yml:4]: Sure updated

**Pull request merged**